docs_src/microservices/security/edgex-secrets-config-proxy.md

```diff
@@ -0,0 +1,140 @@
+
+    * --key _/path/to/private.pem_ (required)
+
+      Path to TLS private key.
+
+  * **adduser**
+
+    Create a proxy user using JWT public/private key authentication. Requires additional arguments:
+
+    * --oauth (unimplemented; reserved for future use)
+
+      Create user using OAuth authentication plugin.
+
+    * --jwt (required)
+
+      Create user using JWT authentication plugin.
+
+    * --algorithm HS256 | ES256 (required for JWT method)
+
+      Algorithm used for signing the JWT.
+
+    * --public\_key _public\_key.pem_ (required for JWT method)
+
+      Public key used to validate the JWT. This key is assumed to have been pre-created using some external mechanism such as a TPM, HSM, openssl, or other method.
+
+    * --user _username_ (required)
+
+      Username of the user to add.
+
+    * --id _key_ (optional)
+
+      Optional user-specified &quot;key&quot; used for linkage with an incoming JWT via Kong&#39;s config.key\_claim\_name setting (defaults to &quot;iss&quot; field).
+
+    * --group _group_ (optional)
+
+      Group to which the user belongs, defaults to &quot;admin&quot;.
+
+    * --output-key _path_ (optional)
+
+      Output auto-generated key to a file (if --id option not supplied).
+
+  * **deluser**
+
+    Delete a proxy user. Requires additional arguments:
+
+    * --oauth (unimplemented; reserved for future use)
+
+      Create user using OAuth authentication plugin.
+
+    * --jwt (required)
+
+      Delete user using JWT authentication plugin.
+
+    * --user _username_ (required)
+
+      Username of the user to delete.
+
+  * **jwt**
+
+    Utility function to create a JWT proxy authentication token from a supplied secret. This command does not require secret store access, but the values supplied must match those presented to the adduser command earlier. Requires additional arguments:
+
+    * --algorithm `HS256` | `ES256` (required)
+
+      Algorithm used for signing the JWT.
+
+    * --id _key_ (required)
+
+      The &quot;key&quot; field from the &quot;adduser&quot; command.
+      (This will be either the --id argument passed in, or the --output-key file contents.)
+
+    * --private\_key _private\_key.pem_ (required)
+
+      Private key used to sign the JWT.
+
+    * --exp _duration_ (optional)
+
+      Duration of generated jwt expressed as a golang-parseable duration value. Use &quot;never&quot; to omit an expiration field in the JWT. Defaults to &quot;1h&quot; (one hour) if unspecified.
+
+    * --output _path_  (recommended but optional)
+
+      Output encoded JWT token to a file.
+
+      The generated JWT will be the encoded representation of:
+
+      <pre>
+      {
+        &quot;typ&quot;: &quot;JWT&quot;,
+        &quot;alg&quot;: &quot;HS256 | ES256&quot;
+      }
+      {
+        &quot;iss&quot;: &quot;_key_&quot;,
+        &quot;exp&quot;: (calculated expiration time)
+      }
+      (signature)
+      </pre>
+
+# ENVIRONMENT
+
+  * **IKM\_HOOK**
```

Comment:
This still isn't clear to me. What  `IKM\_HOOK` is and how to use it.  
Shouldn't need to look at security-secretstore-setup to understand it.

Reply:
Maybe we should note that this hasn't yet been implemented, but is an upcoming feature?